### PR TITLE
feat: Adjusted the hitbox coordinates of where the stones should be dropped (FM-266).

### PR DESCRIPTION
**Pull request by bernhardccodev**
# Changes
- Moved the handling if the stones dropped on the correct coordinates from gameplay-scene.ts to monster.ts.

# How to test
- Run locally, then start the game.
- Select any level and during the gameplay, dropping of stones should be withing or very close to the monster's face.
- Repeat this process on different screen sizes.

Ref: [FM-266](https://curiouslearning.atlassian.net/browse/FM-266)

[FM-266]: https://curiouslearning.atlassian.net/browse/FM-266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/monster.ts [hunk 1]
```diff
@@ -64,6 +62,32 @@ export class Monster extends EventManager {
...
+
+    //Adjust this range factor to control how big is the hit box for dropping stones.
+    const rangeFactorX = 70; //SUBCTRACT FROM CENTER TO LEFT, ADD FROM CENTER TO RIGHT.
+    const rangeFactoryY = 50; //SUBCTRACT FROM CENTER TO TOP, ADD FROM CENTER TO BOTTOM.
+
+    const monsterXCenter = this.game.width / 2;
+    //Note: Rive height is currently always half of width. This might change when new rive files are to be implemented/
+    const monsterYCenter = monsterXCenter / 2; //Create different sets of height for multiple rive files or adjust this for height when replacing the current rive monster.
+    const isWithinHitboxY = y <= (monsterYCenter + rangeFactoryY) && y >= (monsterYCenter - rangeFactoryY);
+    const isWithinHitboxX = x <= (monsterXCenter + rangeFactorX) && x >= (monsterXCenter - rangeFactorX);
```
Comment (miguelccodev) [inline src/components/monster.ts:86]: As a perf improvement,

`(monsterYCenter + rangeFactoryY) && y >= (monsterYCenter - rangeFactoryY);` and  `(monsterXCenter + rangeFactorX) && x >= (monsterXCenter - rangeFactorX);` can be calculated on instantiation.

The app has a reinitialize logic when screen resizes so we only need to determine those once and just compare x and y respectively.

**Review by woodsjohn-codev — APPROVED** [review]
looks good to me

**Review by miguelccodev — CHANGES_REQUESTED** [review]

**Review by miguelccodev — APPROVED** [review]
Looks good!

File: src/components/monster.ts [hunk 2]
```diff
@@ -62,6 +75,37 @@ export class Monster extends EventManager {
     //     this.changeToIdleAnimation(); // Return to idle after any other animation
     //   }
     // });
+
+    //Adjust this range factor to control how big is the hit box for dropping stones.
+    const rangeFactorX = 70; //SUBCTRACT FROM CENTER TO LEFT, ADD FROM CENTER TO RIGHT.
+    const rangeFactoryY = 50; //SUBCTRACT FROM CENTER TO TOP, ADD FROM CENTER TO BOTTOM.
```
Comment (ashwinnair-chimple) [inline src/components/monster.ts:81]: Small correction with naming rangeFactor instead of rangeFactory

File: src/components/monster.ts [hunk 3]
```diff
@@ -62,6 +75,37 @@ export class Monster extends EventManager {
...
+    const monsterCenterX = this.game.width / 2;
+    //Note: Rive height is currently always half of width. This might change when new rive files are to be implemented/
+    const monsterCenterY = monsterCenterX / 2; //Create different sets of height for multiple rive files or adjust this for height when replacing the current rive monster.
+
+    this.hitboxRangeX.from = monsterCenterX - rangeFactorX;
+    this.hitboxRangeX.to = monsterCenterX + rangeFactorX;
+    this.hitboxRangeY.from = monsterCenterY - rangeFactoryY;
```
Comment (ashwinnair-chimple) [inline src/components/monster.ts:88]: Small correction with naming rangeFactorY instead of rangeFactoryY

File: src/components/monster.ts [hunk 4]
```diff
@@ -62,6 +75,37 @@ export class Monster extends EventManager {
...
+    this.hitboxRangeY.to = monsterCenterY + rangeFactorX;
```
Comment (ashwinnair-chimple) [inline src/components/monster.ts:89]: Shouldn't this line of code be 
this.hitboxRangeY.to = monsterCenterY + **rangeFactorY**; instead of **X**

**Review by ashwinnair-chimple — COMMENTED** [review]
added few comments

**Review by ashwinnair-chimple — APPROVED** [review]
Approved

**Pull request merged**